Userland/Libraries/LibX86/Types/Formatter.cpp

```diff
@@ -49,12 +50,13 @@ void AK::Formatter<u128>::format(AK::FormatBuilder& builder, u128 value)
         VERIFY_NOT_REACHED();
     }
 
-    u16 lower_length = sizeof(u64) * 0xFF / base;
-    if (m_width.value() > lower_length) {
-        builder.put_u64(value.high(), base, m_alternative_form, upper_case, m_zero_pad, m_align, m_width.value() - lower_length, m_fill, m_sign_mode);
-        builder.put_u64(value.low(), base, false, upper_case, m_zero_pad, m_align, m_width.value(), m_fill, m_sign_mode);
+    u16 lower_length = sizeof(u64) * 8 / base;
+    auto width = m_width.value_or(0);
+    if (value.high()) {
+        builder.put_u64(value.high(), base, m_alternative_form, upper_case, m_zero_pad, m_align, max(width - lower_length, 0ul), m_fill, m_sign_mode);
+        builder.put_u64(value.low(), base, false, upper_case, m_zero_pad, m_align, width, m_fill, m_sign_mode);
```

Comment:
Please explain what exactly this fixes in the commit message. "Fix" doesn't explain what was fixed and why.

Reply:
So elaborate in the comit message?
Ok will do.

Will be something like:
> In the past we didn't calculate the part_length correctly which led to us only printing the lower half of the Value,
also we took the length argument as granted, which is more than often not the case,
in addition to that the lenght argument isn't treated as a upperbound anymore